angular-ngrx-scss/src/app/auth/services/auth.service.spec.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { environment } from 'src/environments/environment';
+import { SignInResponse, AuthResponse } from '../interfaces/auth';
+import { AuthService } from './auth.service';
+import { TokenService } from './token.service';
+
+describe('AuthService', () => {
+  let tokenService: TokenService;
+  let authService: AuthService;
+  let httpController: HttpTestingController;
+  let token: string | null = 'hello';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: TokenService,
+          useValue: {
+            getToken() {
+              return token;
+            },
+            saveToken(token: string) {
+              return;
+            },
+            removeToken() {
+              return;
+            },
+            removeRefreshToken() {
+              return;
+            },
+          },
+        },
+      ],
+    });
+    tokenService = TestBed.inject(TokenService);
+    authService = TestBed.inject(AuthService);
+    httpController = TestBed.inject(HttpTestingController);
+  });
+
+  it('should redirect the user to the sign in link', () => {
+    const expectedSignInResponse = {
+      redirectUrl: `${environment.apiUrl}/auth/signin`,
+    };
+
+    authService.signIn().subscribe((res) => {
+      expect(res).toEqual(expectedSignInResponse as SignInResponse);
+    });
```

Comment:
to avoid casting you should be able to explicitly type the `expectedSignInResponse`
```suggestion
    const expectedSignInResponse: SignInResponse = {
      redirectUrl: `${environment.apiUrl}/auth/signin`,
    };

    authService.signIn().subscribe((res) => {
      expect(res).toEqual(expectedSignInResponse);
    });
```

Reply:
Will update this when we do the OAuth update!